infra/src/main/kotlin/fi/fta/geoviite/infra/geocoding/Geocoding.kt

```diff
@@ -580,10 +591,17 @@ fun getProjectedAddressPoints(projectionLines: List<ProjectionLine>, alignment:
     while (edgeIndex <= alignmentEdges.lastIndex && projectionIndex <= projectionLines.lastIndex) {
         val edge = alignmentEdges[edgeIndex]
         val projection = projectionLines[projectionIndex]
+        // Check if the edge goes in the same direction as the reference line at projection point
```

Comment:
Tässä oikeastaan se varsinainen logiikka: verrataan projetkioviivan lähteen (pituusmittauslinjan) suuntaa raiteen suuntaan tuon verrattavan edgen kohdalla. Jos ne menee vastakkaisiin suuntiin (yli 90 asteen ero), niin eteen/taakse tulkinnat pitää tehdä toisin päin. Nähdäkseni tämä korjaa zig-zag tilanteet sekä raiteessa että pituusmittauslinjassa.

Algoritmi:
- Tämä käy raiteen ja projektioviivat kerran läpi yhtenä iteraationa, olettaen että ne menee samaan suuntaan (koska niinhän niiden pitäisi). Tuo säästää prosessointia aika paljon verrattuna siihen että kokeiltaisi etsiä kunkin projektion osumakohta itsenäisesti.
- Jos käsiteltävä projektio menee käsiteltävää edgeä edestä ohi, hypätään seuraavaan edgeen. Myöhempiäkään projektioita ei tarvitse enää verrata taaksepäin koska osoitteiden pitää aina kasvaa ja käsiteltävän projektion osoite tulee raiteella vasta myöhemmin.
- Jos käsiteltävä projektio menee käsiteltävän edgen takaa ohi, on käynyt jotain hassua. Kyseinen osoite ei ole tälle raiteelle validi, koska osoitteiden pitää kasvaa koko ajan. Tällöin hypätään seuraavaan projektioon (osoitteeseen) koska tuolla epävalidilla ei tee mitään.
- Jos projektio osuu edgeen, löydettiin tietenkin osoitteen kohta

Muutos:
- Jos pituusmittauslija projektion kohdalla meneekin eri suuntaan kuin raide, on jokin tietenkin vinksallaan. Jommassa kummassa on zig-zag tai sitten koko pituusmittauslinja on vain yhteensopimaton koko raiteelle. Jos koko homma on jojossa niin tämä ei ehkä tuota mitään järkevää mutta ei tuottanut alkuperäinenkään. Eli oletetaan että kyseessä on tilapäinen zigzag ja kokeillaan tuottaa ne osoitteet jotka voidaan.
- Jos projektio menee edgen "edestä" ohi, oikea törmäyskohta onkin lähempänä alkua sillä suunta on väärin päin. Tällöin siis kyseinen osoite on nimenomaa epävalidi koska osoitteiden pitäisi taas kasvaa -> otetaan seuraava projektio.
- Jos projektio menee edge "takaa" ohi, oikea törmäyskohta on raidetta eteenpäin -> kokeillaan seuraava edge.
- Jälleen, jos törmäyskohta on raiteen sisällä niin osoitepiste löytyi. Zig-zagin tapauksessa tuon matchin voisi varmaan myös suodattaa pois, sillä luultavasti zig-zagin jälkeen löytyy vielä toinen match (kun kohta ohitetaan uudelleen), mutta tuo ei liene tarpeen.